handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1359,7 +1366,7 @@ private void rejectRemoteInitiatedRenegotiation() throws SSLHandshakeException {
         // As rejectRemoteInitiatedRenegotiation() is called in a finally block we also need to check if we shutdown
         // the engine before as otherwise SSL.getHandshakeCount(ssl) will throw an NPE if the passed in ssl is 0.
         // See https://github.com/netty/netty/issues/7353
-        if (!isDestroyed() && SSL.getHandshakeCount(ssl) > 1 &&
+        if (!isDestroyed() && !clientMode && SSL.getHandshakeCount(ssl) > 1 &&
```

Comment:
It looks like this would allow unlimited renegotiation for non-boringssl on client-side. Am I reading this right?

Reply:
Yeah, shutdown and throw is skipped when client-side. This looks suspicious.

Reply:
yes ... I guess we could still add some guards. Let me do this